### PR TITLE
[tcgc] default `.access` to `"public"`

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes https://github.com/Azure/typespec-azure/issues/926

### CODE REVIEW

File: docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx [hunk 1]
```diff
@@ -1360,7 +1360,7 @@ export interface SdkModelType extends SdkTypeBase {
   generatedName: string;
   description?: string;
   details?: string;
-  access?: AccessFlags;
+  access: AccessFlags;
```
Comment (m-nash) [inline docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx:1363]: @archerzz Our goal is to take the types from tcgc and directly serialize them into codemodle.json file.  Right now the decorator representation work is not done so you won't be able to serialize this as is and get the information on the other side when you deserialize unless you synthetically add a property `wasAccessOverriden`.  What are your thoughts on doing that in the short term until the decorator work is complete vs having this pr wait for the decorator work to happen first and then we do this?

File: packages/typespec-client-generator-core/src/types.ts [hunk 2]
```diff
@@ -557,7 +557,7 @@ export function getSdkModelWithDiagnostics(
       details: docWrapper.details,
       properties: [],
       additionalProperties: undefined, // going to set additional properties in the next few lines when we look at base model
-      access: undefined, // dummy value since we need to update models map before we can set this
+      access: "public", // dummy value since we need to update models map before we can set this
```
Comment (srnagar) [inline packages/typespec-client-generator-core/src/types.ts:560]: Since this is the default value now, we can change the comment

**Review by srnagar — APPROVED** [review]

**Comment by haolingdong-msft** [comment]
Could we consider the anonymous model used in spread parameter in operation case? In this case, the model should be internal as users cannot see it. e.g.
```
  @post
  sendLong(
    ...LongRequest
  ): OkResponse;
```
Here `SendLongRequest` will be anonymous model and the access should be internal as it is not visible to user.

File: docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx (hunk 1 above)
Comment (archerzz) [reply docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx:1363]: > @archerzz Our goal is to take the types from tcgc and directly serialize them into codemodle.json file. Right now the decorator representation work is not done so you won't be able to serialize this as is and get the information on the other side when you deserialize unless you synthetically add a property `wasAccessOverriden`. What are your thoughts on doing that in the short term until the decorator work is complete vs having this pr wait for the decorator work to happen first and then we do this?

@m-nash I think we can go with a workaround. We can use `__raw` to check if the `@access` is explicitly set.

**Comment by iscai-msft** [comment]
@haolingdong-msft @tadelesh is handling that in [this](https://github.com/Azure/typespec-azure/pull/886) PR

File: docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx (hunk 1 above)
Comment (iscai-msft) [reply docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx:1363]: I think it's fair to use `.__raw` for now, will wait on chat discussion to see what happens

**Review by m-nash — APPROVED** [review]

File: docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx (hunk 1 above)
Comment (m-nash) [reply docs/howtos/DataPlane Generation - DPG/07tcgcTypes.mdx:1363]: @archerzz can you link the follow up issue to stop using __raw and start using .decorators once that is in place so we don't lose track of that?

**Review by joheredi — APPROVED** [review]

**Pull request merged**